### PR TITLE
Fix: Handle malformed tool calls from OpenAI using sanitizer

### PR DESCRIPTION
**Pull request by PranjalPragya**
## Problem 

Previously, when OpenAI returned a malformed tool_calls response — such as when function.arguments was None, an invalid type, or not a JSON string — it would cause the Autogen agent to crash.
This commonly occurred during tool call processing, especially when relying on external model behavior that couldn’t be guaranteed to be clean or well-formed.

## Solution

This PR introduces a new sanitizer in autogen_core/utils/sanitizer.py that:
- Intercepts and inspects each tool call in the list.
- Skips calls with missing or malformed function.arguments.
- Filters out invalid JSON or ill-structured inputs.
- Ensures that only well-formed, decodable tool calls are passed onward for execution.

This enhances Autogen's fault-tolerance, especially during external API usage and multi-agent orchestration where malformed tool responses can silently derail entire chains.

## Testing

A dedicated test (`test_sanitizer.py`) was added to:

- Simulate OpenAI-style tool call outputs, including edge cases.
- Validate that malformed entries are removed while valid ones remain unaffected.
- Ensure that the sanitizer logic works in isolation, independent of OpenAI API availability.

## Notes

- This PR does not modify agent or model logic— the sanitizer operates as a standalone pre-execution guard.
- Other test failures (if any) are unrelated and existed before this change.

## Checks

- [ ] I've included any doc changes needed for <https://microsoft.github.io/autogen/>.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by PranjalPragya** [comment]
@microsoft-github-policy-service agree

File: python/packages/autogen-ext/src/autogen_ext/models/openai/_openai_client.py [hunk 1]
```diff
@@ -673,14 +672,20 @@ async def create(
 
         if cancellation_token is not None:
             cancellation_token.link_future(future)
+
         result: Union[ParsedChatCompletion[BaseModel], ChatCompletion] = await future
         if create_params.response_format is not None:
             result = cast(ParsedChatCompletion[Any], result)
+        import sys 
+        import os 
+        sys.path.append(os.path.abspath("python/packages/autogen-core/src"))
```
Comment (ekzhu) [inline python/packages/autogen-ext/src/autogen_ext/models/openai/_openai_client.py]: Please move the import to the top, and not adding system path

Comment (PranjalPragya) [reply python/packages/autogen-ext/src/autogen_ext/models/openai/_openai_client.py]: Thanks @ekzhu, I've removed the sys.path modification and moved the import to the top. Let me know if any other changes are needed!

**Comment by PranjalPragya** [comment]
@microsoft-github-policy-service agree

**Comment by PranjalPragya** [comment]
Hi @ekzhu, just following up on this PR — I’ve made the requested changes and confirmed all CLA requirements are now satisfied. Let me know if there's anything else needed from my side. Thanks!

**Comment by PranjalPragya** [comment]
Hi again @ekzhu, just following up in case this PR got buried. Happy to rebase or resolve conflicts if needed — just let me know. Thanks again for reviewing earlier!

**Comment by tejas-dharani** [comment]
Hi @PranjalPragya,

Could you please resolve the merge conflicts with the main branch?

Thanks!

**Comment by PranjalPragya** [comment]
Hi @ekzhu / @tejas-dharani,

I’ve resolved the merge conflicts with the main branch and committed the changes. All CLA requirements are satisfied, and all requested modifications have been applied. Please let me know if anything else is needed from my side for review.

Thanks!
